src/common/transformations/src/transformations/mlir/mlir_op.cpp

```diff
@@ -267,6 +290,138 @@ namespace mlir {
 
 using namespace ::mlir;
 
+std::shared_ptr<MLIREvaluateBase> MLIREvaluateBase::create(OwningOpRef<ModuleOp> module,
+                                                           MlirMode mode,
+                                                           std::shared_ptr<ov::EvaluationContext> loweringContext) {
+    switch (mode) {
+        #if defined(GRAPH_COMPILER) && defined(GC_ENABLE_IMEX)
+        case MLIR_MODE_GC_GPU:
+            return std::make_shared<MLIREvaluateGcGPU>(std::move(module), loweringContext);
+        #endif
+        case MLIR_MODE_TPP:
+        case MLIR_MODE_GC:
+        case MLIR_MODE_DEFAULT:
+            return std::make_shared<MLIREvaluate>(std::move(module), mode);
+        default:
+            OPENVINO_THROW("Unsupported MLIR mode");
+    }
+}
+
+#if defined(GRAPH_COMPILER) && defined(GC_ENABLE_IMEX)
```

Comment:
```suggestion
#ifdef GC_USE_IMEX
```

Reply:
I think it's sufficient to just check GC_USE_IMEX here and in all other cases below.